File: EMPLOYEEAPI-BACKEND/src/main/java/com/klef/jenkinsfsd/service/EmployeeService.java
package com.klef.jenkinsfsd.service;

import com.klef.jenkinsfsd.entity.Employee;
import java.util.List;

public interface EmployeeService {
    List<Employee> getAllEmployees();
    Employee getEmployeeById(int id);
    Employee addEmployee(Employee employee);
    Employee updateEmployee(int id, Employee employee);
    void deleteEmployee(int id);
}

File: EMPLOYEEAPI-REACT/employeeapi-react/src/services/EmployeeService.js
import axios from "axios";

const EMPLOYEE_API_BASE_URL = "http://localhost:2001/api/employees";

class EmployeeService {
  
  getAllEmployees() {
    return axios.get(`${EMPLOYEE_API_BASE_URL}/all`);
  }

  getEmployeeById(id) {
    return axios.get(`${EMPLOYEE_API_BASE_URL}/get/${id}`);
  }

  addEmployee(employee) {
    return axios.post(`${EMPLOYEE_API_BASE_URL}/add`, employee);
  }

  updateEmployee(id, employee) {
    return axios.put(`${EMPLOYEE_API_BASE_URL}/update/${id}`, employee);
  }

  deleteEmployee(id) {
    return axios.delete(`${EMPLOYEE_API_BASE_URL}/delete/${id}`);
  }
}

export default new EmployeeService();
